phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3524,14 +3525,26 @@ private void mutateStringProperty(String tenantId, String schemaName, String tab
 
     public MutationState addColumn(AddColumnStatement statement) throws SQLException {
         PTable table = FromCompiler.getResolver(statement, connection).getTables().get(0).getTable();
-        return addColumn(table, statement.getColumnDefs(), statement.getProps(), statement.ifNotExists(), false, statement.getTable(), statement.getTableType());
+        return addColumn(table, statement.getColumnDefs(), statement.getProps(), statement.ifNotExists(), false, statement.getTable(), statement.getTableType(), statement.isCascade(), statement.getIndexes());
     }
 
     public MutationState addColumn(PTable table, List<ColumnDef> origColumnDefs,
             ListMultimap<String, Pair<String, Object>> stmtProperties, boolean ifNotExists,
-            boolean removeTableProps, NamedTableNode namedTableNode, PTableType tableType)
+            boolean removeTableProps, NamedTableNode namedTableNode, PTableType tableType, boolean cascade, List<NamedNode> indexes)
                     throws SQLException {
         connection.rollback();
+        if (cascade && (indexes == null || indexes.size()>0)) {
```

Comment:
The condition (indexes==null || indexes.size()>0) seems very un-intuitive. It requires folks to understand that indexes==null is the case with ALL and the latter is when there is explicit list of indexes; which is not very clear just reading this code. 
I'd just remove this condition entirely for now. It is not relevant for the throwing NOT_SUPPORTED_CASCADE_FEATURE exception for views. And indexes should be properly handled when the feature is implemented.

Reply:
You are right! On a second thought, I agree with you. Ll keep the condition simple for the base PR.